resbundler.js

```diff
@@ -0,0 +1,147 @@
+      "-c or --compiled\n" +
+      "  Whether you want the output to be compiled with uglify-js. Valid values are 'compiled', and 'uncompiled'. Default: 'compiled'.");
+      "-r or --resDir\n" +
+      "  directory to read and file json files. Default: resources" +
+      "-o or --outDir\n" +
+      "  directory to place output files. Default: current dir." +
+      "-f or --filename\n" +
+      "  specify output file name when assembly mode is assembled. Default: ilib-translation.js" +
+      "-l or --locales\n" +
+      "  Locales you want your make js files when assembely mode is `dynamic`. Value is a comma-separated list of BCP-47 style locale tags.";
+    process.exit();
+}
+
+var argv = process.argv;
+var options = {
+    assembly: "assembled",
+    compiled: "compiled",
+    resDir: "resources",
+    outDir: ".",
+    filename: "ilib-translation.js",
+    locales: []
+}
+var result = "";
+
+for (var i = 0; i < argv.length; i++){
+    var val = argv[i];
+    if (val === "-h" || val == "--help"){
+        usage();
+    } else if (val === "-a" || val === "--assembly"){
+        options.assembly = argv[++i];
+    } else if(val === "-c" || val === "--compiled"){
+        options.compiled = argv[++i];
+    } else if (val === "-r" || val === "--resDir"){
+        options.resDir = argv[++i];
+    } else if (val === "-o" || val === "--outDir"){
+        options.outDir = argv[++i];
+        if (!fs.existsSync(options.outDir)){
+            fs.mkdirSync(options.outDir);
+        }
+    } else if (val === "-f" || val === "--filename"){
+        options.filename = argv[++i];
+    } else if (val === "-l" || val === "--locales"){
+        if (i < argv.length && argv[i+1]) {
+            options.locales = argv[++i].split(",");
+        }
+    }
+}
+
+function manipulateKey(fullPath){
+    if (!fullPath) return;
+    var prefix = "ilib.data.strings";
+
+    var splitPath = fullPath.split("/");
+    var key = splitPath.slice(1, splitPath.length -1).join("_");
+    key = prefix + (key != "" ? "_" + key : key);
+    return key;
+}
+
+function assemblFiles(dir){
+    var list = fs.readdirSync(dir);
+    list.forEach(function(file){
+        var fullPath = path.join(dir, file);
+        var stat = fs.statSync(fullPath);
+        if (stat && stat.isDirectory()){
+            assemblFiles(fullPath);
+        } else {
+            if (file === "strings.json"){
+                var data = fs.readFileSync(fullPath, "utf-8");
+                var dataKey = manipulateKey(fullPath);
+                if (data) {
+                    var temp = dataKey +  " = " + data + ";\n";
+                    result += temp;
+                }
+            }
+        }
+    });
+    return result;
+}
+
+function dynamicFiles(){
+    if ((options.assembly != "assembled" || options.assembly == "dynamic") && options.locales.length > 0){
+        options.locales.forEach(function(lo){
+            result = "";
+            lo.split("-").map(function(v, i, a){
```

Comment:
Use ilib or ilib-locale to parse the locale spec.
Also, better variable names would be good. What do "v", "i" , and "a" mean?

Reply:
I updated them. and fixed typo.